dev-tools/generate_notice.py

```diff
@@ -51,35 +51,92 @@ def gather_dependencies(vendor_dirs, overrides=None):
 
         # walk looking for LICENSE files
         for root, dirs, filenames in os.walk(vendor):
-            for filename in sorted(filenames):
-                if filename.startswith("LICENSE") and "docs" not in filename:
-                    lib_path = get_library_path(root)
-                    lib_search = [l for l in libs if l["path"].startswith(lib_path)]
-                    if len(lib_search) == 0:
-                        print("WARNING: No version information found for: {}".format(lib_path))
-                        lib = {"path": lib_path}
-                    else:
-                        lib = lib_search[0]
-                    lib["license_file"] = os.path.join(root, filename)
-
-                    lib["license_contents"] = read_file(lib["license_file"])
-                    lib["license_summary"] = detect_license_summary(lib["license_contents"])
-                    if lib["license_summary"] == "UNKNOWN":
-                        print("WARNING: Unknown license for: {}".format(lib_path))
-                    revision = overrides.get(lib_path, {}).get("revision")
-                    if revision:
-                        lib["revision"] = revision
-                    if lib_path not in dependencies:
-                        dependencies[lib_path] = [lib]
-                    else:
-                        dependencies[lib_path].append(lib)
+            licenses = get_licenses(root)
+            for filename in licenses:
+                lib_path = get_library_path(root)
+                lib_search = [l for l in libs if l["path"].startswith(lib_path)]
+                if len(lib_search) == 0:
+                    print("WARNING: No version information found for: {}".format(lib_path))
+                    lib = {"path": lib_path}
+                else:
+                    lib = lib_search[0]
+                lib["license_file"] = os.path.join(root, filename)
+
+                lib["license_contents"] = read_file(lib["license_file"])
+                lib["license_summary"] = detect_license_summary(lib["license_contents"])
+                if lib["license_summary"] == "UNKNOWN":
+                    print("WARNING: Unknown license for: {}".format(lib_path))
```

Comment:
For the future: We could exit with 1 in such cases to make sure CI fails and we don't add any unknown licenese.

Reply:
There's another check for "acceptable" licenses at the end of the script, which raises an exception and therefore makes CI red.